### DISCUSSION TITLE
Bugfix: HDF5 in MET 11.1.0 does not compile with GNU 11

### DISCUSSION THREAD

**Discussion by HathewayWill** [dtcenter/METplus#2303] [Installation] [answered]
## Describe the Problem ##
The GNU compilers fail to build MET using GNU 11.4.0

### Expected Behavior ###
Change from MET 11.0.2 to 11.1.0 should work without any issues

### Environment ###
Describe your runtime environment:
*1. Machine: Linux Workstation
*2. OS: Ubuntu 22.04.2
*3. Software version number(s)*

TEST_BASE = /home/workhorse/DTC/MET-11.1.0
MET_SUBDIR = /home/workhorse/DTC/MET-11.1.0
MET_TARBALL = v11.1.0.tar.gz
USE_MODULES = FALSE
MAKE_ARGS = -j
LD_LIBRARY_PATH = 

Compiling libraries into: /home/workhorse/DTC/MET-11.1.0/external_libs
COMPILER = gnu_11.4.0
COMPILER_FAMILY = gnu
COMPILER_VERSION = 11.4.0

USE_MODULES = FALSE

export  CC=/usr/bin/gcc
export CXX=/usr/bin/g++
export  FC=/usr/bin/gfortran
export F77=/usr/bin/gfortran
export F90=/usr/bin/gfortran

Using python version: 
Python 3.10.12

### To Reproduce ###

see attached script and error logs

[New Folder.tar.gz](https://github.com/dtcenter/MET/files/12234791/New.Folder.tar.gz)

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Organization** level **Project** for support of the current coordinated release
- [ ] Select **Repository** level **Project** for development toward the next official release or add **alert: NEED CYCLE ASSIGNMENT** label
- [ ] Select **Milestone** as the next bugfix version

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Bugfix Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [ ] Fix the bug and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Organization** level software support **Project** for the current coordinated release
Select: **Milestone** as the next bugfix version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
Select: **Reviewer(s)** and **Development** issue
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Close this issue.

**Comment by jprestop** [comment, marked as answer]
@HathewayWill, I think the problem may lie in your environment.  I was able to successfully compile HDF5 and MET 11.1.0 on my Mac.  I took a quick look at your log file and did not immediately identify the problem.  I'll try to take a closer look soon.  In the meantime, I've attached my log files in order for you to take a look and with hope that perhaps you might be able to identify what is different about your environment.  Please let us know if you have any questions.
[gcc11.4_files.tar.gz](https://github.com/dtcenter/MET/files/12244589/gcc11.4_files.tar.gz)

**Reply by jprestop** [reply]
@HathewayWill Since I was able to successfully compile HDF5 and MET 11.1.0 on my Mac, so I'm going to go ahead and close this Discussion.  Please feel free to post again if you find you cannot successfully compile MET 11.1.0.
